ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/transaction/UtxoLedgerService.kt

```diff
@@ -0,0 +1,60 @@
+package net.corda.v5.ledger.utxo.transaction
+
+import net.corda.v5.base.annotations.DoNotImplement
+import net.corda.v5.base.annotations.Suspendable
+import net.corda.v5.ledger.common.transaction.Party
+import net.corda.v5.ledger.utxo.ContractState
+import net.corda.v5.ledger.utxo.StateAndRef
+import net.corda.v5.ledger.utxo.StateRef
+
+/**
+ * Defines UTXO ledger services.
+ */
+@DoNotImplement
+interface UtxoLedgerService {
+
+    /**
+     * Gets a UTXO transaction builder
+     *
+     * @param notary The notary associated with the transaction builder.
+     * @return Returns a new [UtxoTransactionBuilder] instance.
+     */
+    @Suspendable
+    fun getTransactionBuilder(notary: Party): UtxoTransactionBuilder
```

Comment:
@mnesbit Since all transactions need a notary, it's better that it's not declared nullable. It should be trivial to obtain a notary from either an underlying service, or a pre-existing state before creating a transaction builder.

Reply:
@mnesbit Lots of builder patterns allow you to pass initial data via a constructor (which is essentially what this is doing). `TransactionBuilder` in Corda 4 allowed you to pass the notary via the constructor. `StringBuilder` allows you to pass in an initial string `val x = StringBuilder("Hello, World!")`.

The only way I can think of having a `setNotary` function that aligns with `setTimeWindow` would be to default it at creation, which would then need service access to get the default notary (and I'm not even sure that's useful to the developer). For time window, we could have a `DefaultTimeWindowImpl` where `from` is `Instant.MIN` and `until` is `Instant.MAX`, which would give the equivalent behaviour to Corda 4 with a `null` time window.

Reply:
Lets just go with setNotary on builder. I think it is just nicer and my quick poll of Rick's view was similar